EDDI/EDDI.cs

```diff
@@ -1005,68 +1016,62 @@ private bool eventLocation(LocationEvent theEvent)
                 }
             }
 
-            if (theEvent.docked == true || theEvent.bodytype.ToLowerInvariant() == "station")
+            if (theEvent.docked || theEvent.bodytype.ToLowerInvariant() == "station")
             {
-                // In this case body === station and our body information is invalid
+                // In this case body = station and our body information is invalid
                 CurrentStellarBody = null;
 
-                // Force first location update even if it matches with 'firstLocation' bool
-                if (!firstLocation && (CurrentStation != null && CurrentStation.name == theEvent.station))
-                {
-                    // We are already at this station; nothing to do
-                    Logging.Debug("Already at station " + theEvent.station);
-                    return false;
-                }
-                firstLocation = false;
-
                 // Update the station
-                Logging.Debug("Now at station " + theEvent.station);
-                Station station = CurrentStarSystem.stations.Find(s => s.name == theEvent.station);
+                string stationName = theEvent.docked ? theEvent.station : theEvent.body;
+
+                Logging.Debug("Now at station " + stationName);
+                Station station = CurrentStarSystem.stations.Find(s => s.name == stationName);
                 if (station == null)
                 {
                     // This station is unknown to us, might not be in our data source or we might not have connectivity.  Use a placeholder
                     station = new Station
                     {
-                        name = theEvent.station,
+                        name = stationName,
                         systemname = theEvent.system
                     };
```

Comment:
Add this station to our `CurrentStarSystem`?

Reply:
That's legacy code repeated throughout EDDI.cs. . Not sure why you'd want to add a 'placeholder' to a StarSystem object.

Reply:
The idea would be to scoop up the data and make it into something that we can reference, then replace it once more complete data was available.

Reply:
Fair. A "Well, this is the way it's always been" response is obviously not logical or reasonable.

Reply:
Done.